go/netutil/netutil_test.go

```diff
@@ -66,9 +64,7 @@ func TestJoinHostPort(t *testing.T) {
 		"[::1]:321":     {host: "::1", port: 321},
 	}
 	for want, input := range table {
-		if got := JoinHostPort(input.host, input.port); got != want {
-			t.Errorf("SplitHostPort(%v, %v) = %#v, want %#v", input.host, input.port, got, want)
-		}
+		assert.Equal(t, want, JoinHostPort(input.host, input.port))
```

Comment:
outside the scope of this PR, but should we delete this function in favor of the stdlib `net.JoinHostPort`? cc @mattlord, the signature is slightly different (we take an `int32` port whereas stdlib takes `string`)